src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java

```diff
@@ -0,0 +1,719 @@
+/*
+ * Copyright @ 2015 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.impl.neomedia.rtp.remotebitrateestimator;
+
+
+import org.ice4j.util.RateStatistics;
+import org.jitsi.impl.neomedia.rtp.TransportCCEngine;
+import org.jitsi.impl.neomedia.transform.*;
+import org.jitsi.service.neomedia.*;
+import org.jitsi.service.neomedia.rtp.*;
+import org.jitsi.util.*;
+import org.jitsi.util.concurrent.RecurringRunnable;
+import org.jitsi.util.function.Predicate;
+
+import java.util.*;
+
+/**
+ * webrtc.org abs_send_time implementation as of June 26, 2017.
+ * commit ID: 23fbd2aa2c81d065b84d17b09b747e75672e1159
+ * @author Julian Chukwu
+ */
+public class RemoteBitrateEstimatorAbsSendTime
+    extends SinglePacketTransformer
+    implements RemoteBitrateEstimator,
+        RecurringRunnable
+{
+    private static final Logger logger = Logger
+            .getLogger(RemoteBitrateEstimatorAbsSendTime.class);
+
+    /**
+     * Ctor.
+     * <p>
+     * XXX At some point ideally we would get rid of this ctor and all the
+     * inheritors will use the parametrized ctor. Also, we might want to move
+     * this check inside the <tt>TransformEngineChain</tt> so that we only make
+     * the check once per packet: The RTCP transformer is only supposed to
+     * (reverse) transform RTCP packets and the RTP transformer is only
+     * supposed to modify RTP packets.
+     */
+    public RemoteBitrateEstimatorAbsSendTime() {
+        super();
+    }
+
+    /**
+     * Ctor.
+     *
+     * @param packetPredicate the <tt>PacketPredicate</tt> to use to match
+     *                        packets to (reverse) transform.
+     */
+    public RemoteBitrateEstimatorAbsSendTime(Predicate<ByteArrayBuffer> packetPredicate) {
+        super(packetPredicate);
```

Comment:
Why do we need this constructor? The RBE transformer only works with RTP packets, so it should always call super(RTPPacketPredicate.INSTANCE)